examples/ewald_total.cpp

```diff
@@ -556,37 +559,32 @@ std::vector<Real> evaluate_short_range(const TestCaseSystem<Real> &System, Short
 template <typename Real>
 std::vector<Real> compute_green_func(int N, Real alpha, Real L) {
 
+    // TODO: generalize to different box lengths
     const Real h = L / N;
     const Real TWOPI_L = 2 * M_PI / L;
-    const Real FOUR_PI = 4 * M_PI;
-    const Real DENOM = 4 * alpha * alpha;
 
     std::vector<Real> G(N * N * N, 0.0);
-    std::vector<int> k(N);
-    std::vector<int> ksq(N);
-
-    for (size_t i = 0; i < N; ++i) {
-        const int i_new = (i > (N / 2)) ? i - N : i;
-        k[i] = TWOPI_L * i_new;
-        // ksq[i] = k[i] * k[i];
-    }
-
+#pragma omp parallel for default(none) shared(h, TWOPI_L, N, alpha, G)
     for (size_t w = 0; w < N; ++w) {
         for (size_t j = 0; j < N; ++j) {
             for (size_t i = 0; i < N; ++i) {
                 // TODO: optimize this step; structure the loop accordingly
                 // conditionals are expensive (?)
-                const int k_x = k[i];
-                const int k_y = k[j];
-                const int k_z = k[w];
+                const int i_new = (i > (N / 2)) ? i - N : i;
+                const int j_new = (j > (N / 2)) ? j - N : j;
+                const int w_new = (w > (N / 2)) ? w - N : w;
+
+                const int k_x = TWOPI_L * i_new;
+                const int k_y = TWOPI_L * j_new;
+                const int k_z = TWOPI_L * w_new;
```

Comment:
The compiler will do this automatically almost certainly, but you should do some common subexpression elimination yourself by moving N/2 into a const outside the triple loop.